src/tcp/tcp.ts

```
@@ -21,14 +20,25 @@ module Tcp {
     UNKOWN
   }
 
+  // A limit on the max number of TCP connections before we start rejecting
+  // new ones.
+  var DEFAULT_MAX_CONNECTIONS = 1048576;
+
   // Helper function.
-  function endpointOfSocketInfo(info:TcpLib.SocketInfo) : Net.Endpoint {
+  function endpointOfSocketInfo(info:freedom_TcpSocket.SocketInfo)
+      : Net.Endpoint {
      return { address: info.peerAddress, port: info.peerPort }
   }
 
-  // A limit on the max number of TCP connections before we start rejecting
-  // new ones.
-  var DEFAULT_MAX_CONNECTIONS = 1048576;
+  // A static helper function to close a freedom socket object and then
+  // appropriately deallocate it's interface object.
+  function closeFreedomSocket_(socket:freedom_TcpSocket.Socket,
+      idInfoString?:string) : Promise<void> {
+    return socket.close().then(() => {
```

As discussed, I didn't mean we should swallow the exception: we should call `freedom['core.tcpsocket'].close(socket);` even if the call to `socket.close()` rejects (and then reject with the original error).

Like a finally block.